components/suggest/src/db.rs

```diff
@@ -805,6 +806,62 @@ impl<'a> SuggestDao<'a> {
         )
     }
 
+    /// Fetches exposure suggestions
+    pub fn fetch_exposure_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
+        // A single exposure suggestion can be spread across multiple remote
+        // settings records, for example if it has very many keywords. On ingest
+        // we will insert one row in `exposure_custom_details` and one row in
+        // `suggestions` per record, but that's only an implementation detail.
+        // Logically, and for consumers, there's only ever at most one exposure
+        // suggestion with a given exposure suggestion type.
+        //
+        // Why do insertions this way? It's how other suggestions work, and it
+        // lets us perform relational operations on suggestions, records, and
+        // keywords. For example, when a record is deleted we can look up its ID
+        // in `suggestions`, join the keywords table on the suggestion ID, and
+        // delete the keywords that were added by that record.
+
+        let mut suggestions = Vec::new();
+        if let Some(suggestion_types) = query
+            .provider_constraints
+            .as_ref()
+            .and_then(|c| c.exposure_suggestion_types.as_ref())
+        {
+            for suggestion_type in suggestion_types {
```

Comment:
How many types to you expect to be enabled at once?  If it's more than a couple, maybe we could optimize this to select all the types in one query.

Reply:
Only one at a time actually, but I don't think there's a good reason to enforce that in the API since supporting more than one is easy, and who knows, we might want to do more than one at a time someday. Thanks for pointing this out, it should be done in one query.

Reply:
@bendk Could you give my changes to this method (`fetch_exposure_suggestions`) another look please? It's changed substantially and I want to make sure it's OK.

Reply:
These changes look good to me and more importantly the tests in `store.rs` look very complete which is great.